deployment/dockerdeploy/ghcrimageprovider.go

```diff
@@ -74,33 +74,21 @@ func (p *GhcrImageProvider) GetImage(ctx context.Context, def *ImageDef) (*Image
 			return nil, errors.New("cannot pull community edition of columnar")
 		}
 
-		ghcrImagePath := fmt.Sprintf("ghcr.io/cb-vanilla/columnar:%s", serverVersion)
-		image, err := MultiArchImagePuller{
-			Logger:       p.Logger,
-			DockerCli:    p.DockerCli,
-			RegistryAuth: p.genGhcrAuthStr(),
-			ImagePath:    ghcrImagePath,
-		}.Pull(ctx)
-		if err != nil {
-			p.Logger.Debug("ghcr provider failed to provide image from columnar repo", zap.Error(err))
-
-			ghcrImagePath := fmt.Sprintf("ghcr.io/cb-vanilla/couchbase-columnar:%s", serverVersion)
-			image, err := MultiArchImagePuller{
-				Logger:       p.Logger,
-				DockerCli:    p.DockerCli,
-				RegistryAuth: p.genGhcrAuthStr(),
-				ImagePath:    ghcrImagePath,
-			}.Pull(ctx)
-			if err != nil {
-				p.Logger.Debug("ghcr provider failed to provide image from couchbase-columnar repo", zap.Error(err))
-
-				return nil, errors.New("failed to pull image from both columnar and couchbase-columnar repos")
+		var imagePaths []string
+		if len(serverVersion) > 0 && (serverVersion[0] == '0' || serverVersion[0] == '1') {
+			// For versions starting with 0 or 1, try the legacy columnar paths
+			imagePaths = []string{
+				fmt.Sprintf("ghcr.io/cb-vanilla/columnar:%s", serverVersion),
+				fmt.Sprintf("ghcr.io/cb-vanilla/couchbase-columnar:%s", serverVersion),
+			}
+		} else {
+			// For newer versions, use enterprise-analytics
```

Comment:
For this case with newer versions, can we use the normal (non-multi-path) logic as is used for Server deployments?

Reply:
Can do